src/brevitas/graph/equalize.py

```diff
@@ -245,55 +331,100 @@ def _cross_layer_equalization(
         if isinstance(module, nn.MultiheadAttention):
             srcs[i] = module.out_proj
         src_axes[srcs[i]] = _get_output_axis(module)
+        act_sources_axes[srcs[i]] = _get_act_axis(module)
 
     for i, module in enumerate(sinks):
         # If module is not supported, do not perform graph equalization
         if not isinstance(module, _supported_layers):
             return torch.tensor(1., dtype=dtype, device=device)
         # For MultiheadAttention, we support only self-attetion
-        if isinstance(module, nn.MultiheadAttention) and hasattr(sinks[i], 'in_proj_weight'):
+        if isinstance(module, nn.MultiheadAttention) and module.in_proj_weight is not None:
             # For sinks, we only need to modify the weight but not the bias
             sinks[i] = WeightBiasTuple(weight=module.in_proj_weight)
-        elif isinstance(module, nn.MultiheadAttention) and not hasattr(sinks[i], 'in_proj_weight'):
+        elif isinstance(module, nn.MultiheadAttention) and module.in_proj_weight is None:
             return torch.tensor(1., dtype=dtype, device=device)
         sink_axes[sinks[i]] = _get_input_axis(module)
+        act_sink_axes[sinks[i]] = _get_act_axis(module)
+
+    # If act_val is enabled, use source or sink weights to determine the activation channel
```

Comment:
Sink weights are used immediately after we define scale_fn.
After computing sinks_range, we use the srcs weights in case of weight equalization or the activation runtime values in case of act equalization to determine the srcs_range.